span-normalizer/helm/templates/span-normalizer-config.yaml

```diff
@@ -57,6 +57,10 @@ data:
       spanDropCriterion = {{ .Values.spanNormalizerConfig.processor.spanDropCriterion | toJson }}
       {{- end }}
 
+      {{- if hasKey .Values.spanNormalizerConfig.processor "spanDropFilters" }}
+      spanDropFilters = {{ .Values.spanNormalizerConfig.processor.spanDropFilters | indent 4 | trim }}
```

Comment:
toJson not required here?

Reply:
I am not loading as JSON, directly reading as a config object. See this example - https://github.com/hypertrace/hypertrace-ingester/blob/main/span-normalizer/span-normalizer/src/test/resources/configs/span-normalizer/application.conf#L30

I observed that other config were done as toJson, were they read as string and converted to config?

Reply:
But the value that you are providing in the config is a json, this indent and trim function applies to string

Reply:
As shown in the description,  I was using the marker `|-` in YAML that is multi-string notation. So, the entire thing was treated as a big string, and If I apply the `toJson` function to it, it jsonify the string. As an example below.

Input from the above PR description:
```
processor:
    spanDropFilters: |-
      [
        [
          {
            "tagKey": "http.method",
            "operator": "EQ",
            "tagValue": "GET"
          },
          {
            "tagKey": "http.url",
            "operator": "CONTAINS",
            "tagValue": "health"
          }
        ],
        [
          {
            "tagKey": "grpc.url",
            "operator": "NEQ",
            "tagValue": "Sent.TestServiceGetEchos"
          }
        ]
      ]
```

Output with `toJson` on multi-line string using YAML marker `|-`:
```
processor {
      spanDropFilters = "[\n  [\n    {\n      \"tagKey\": \"http.method\",\n      \"operator\": \"EQ\",\n      \"tagValue\": \"GET\"\n    },\n    {\n      \"tagKey\": \"http.url\",\n      \"operator\": \"CONTAINS\",\n      \"tagValue\": \"health\"\n    }\n  ],\n  [\n    {\n      \"tagKey\": \"grpc.url\",\n      \"operator\": \"NEQ\",\n      \"tagValue\": \"Sent.TestServiceGetEchos\"\n    }\n  ]\n]"
    }
```    
And, so there was no need for `toJson` function. However, I modified the code to not use `|-` multi-string marker, and treat the value as a YAML for spanDropFilters. As you guys pointed out, this will be better to catch any config error by converting it to `toJson`, so going it with.

So, the input will be:
```
processor:
    spanDropFilters:
      [
        [
          {
            "tagKey": "http.method",
            "operator": "EQ",
            "tagValue": "GET"
          },
          {
            "tagKey": "http.url",
            "operator": "CONTAINS",
            "tagValue": "health"
          }
        ],
        [
          {
            "tagKey": "grpc.url",
            "operator": "NEQ",
            "tagValue": "Sent.TestServiceGetEchos"
          }
        ]
      ]
```

Output for above using: `{{ .Values.spanNormalizerConfig.processor.spanDropFilters | toJson }}`
```
processor {
      spanDropFilters = [[{"operator":"EQ","tagKey":"http.method","tagValue":"GET"},{"operator":"CONTAINS","tagKey":"http.url","tagValue":"health"}],[{"operator":"NEQ","tagKey":"grpc.url","tagValue":"Sent.TestServiceGetEchos"}]]
    }
```

Reply:
oh ok `|-` this was present